src/rust/ensogl/src/animation/easing.rs

```diff
@@ -245,16 +248,50 @@ impl<T:Value,F,Cb> Animator<T,F,Cb> where F:AnyFnEasing, Cb:Callback<T> {
         self.start();
     }
 
-    /// Stop the animation, rewinds it to the initial value and calls the callback.
-    pub fn rewind(&self) {
+    /// Stop the animation, rewinds it to the provided value and calls the callback.
+    pub fn stop_and_rewind_to(&self, value:T) {
         self.stop();
-        (self.data.callback)(self.start_value());
+        self.set_start_value_no_restart(value);
+        self.data.value.set(value);
+        (self.data.callback)(value);
+    }
+
+    /// Stop the animation, rewinds it to the initial value and calls the callback.
+    pub fn stop_and_rewind(&self) {
+        self.stop_and_rewind_to(self.start_value());
     }
 
     /// Checks whether the animator is running.
     pub fn active(&self) -> bool {
         self.data.active.get()
     }
+
+    /// Restarts the animation using the current value as the new start value.
+    pub fn from_now_to(&self, tgt:T) {
+        let current = self.value();
+        self.data.start_value.set(current);
+        self.data.target_value.set(tgt);
+        if current != tgt {
+            (self.data.callback)(self.start_value());
+            self.reset();
+        }
+    }
+
+    /// Sets the new target value. In case the target value is different then the current target
+    /// value, the animation will be restarted with the current value being the new start value.
```

Comment:
```suggestion
    /// Set the new target value. In case the target value is different from the current target
    /// value, the animation will be restarted with the current value being the new start value.
```